pkg/backend/httpstate/state.go

```diff
@@ -136,15 +139,33 @@ func (u *cloudUpdate) Complete(status apitype.UpdateStatus) error {
 	return u.backend.client.CompleteUpdate(u.context, u.update, status, token)
 }
 
+// recordEvent will record the event with the Pulumi service, enabling things like viewing
+// the rendered update logs or drilling into the timeline of an update.
 func (u *cloudUpdate) recordEvent(
 	action apitype.UpdateKind, event engine.Event, seen map[resource.URN]engine.StepEventMetadata,
 	opts display.Options) error {
+	contract.Assert(u.tokenSource != nil)
+	token, err := u.tokenSource.GetToken()
+	if err != nil {
+		return err
+	}
 
-	// If we don't have a token source, we can't perform any mutations.
-	if u.tokenSource == nil {
-		return nil
+	// Send the event to the Pulumi Service to power things like the update summary page.
+	// Currently opt-in via flag to allow for gathering per data before the service-side changes
+	// are available in production.
+	if os.Getenv("PULUMI_RECORD_ENGINE_EVENTS") != "" {
```

Comment:
NIT: We have `cmdutil.IsTruthy` for these sort of checks

Reply:
Done.